sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   *
+   * The state is null if current status is IDLE.
+   */
+  class State(val state: ViewportState?) : ViewportStatus() {
```

Comment:
Also we should probably make the constructor `internal` so that we can keep adding fields if needed. Developers are not expected to create statuses anyway, are they?

Reply:
-1. I think developers may want to be able to construct these in tests to use as expected values.

Reply:
yeah, agree these could be useful for testing purposes.